static/app/views/projectsDashboard/index.tsx

```diff
@@ -81,6 +82,64 @@ function addProjectsToTeams(teams: Team[], projects: Project[]): TeamWithProject
   }));
 }
 
+function getFilteredProjectsBasedOnTeams({
+  allTeams,
+  userTeams,
+  selectedTeams,
+  isAllTeams,
+  showNonMemberProjects,
+  projects,
+  projectQuery,
+}: {
+  allTeams: Team[];
+  isAllTeams: boolean;
+  projectQuery: string;
+  projects: Project[];
+  selectedTeams: string[];
+  showNonMemberProjects: boolean;
+  userTeams: Team[];
+}): Project[] {
+  const myTeamIds = new Set(userTeams.map(team => team.id));
+
+  const [myTeamsInAll, otherTeamsInAll] = partition(allTeams, team =>
+    myTeamIds.has(team.id)
+  );
+
+  const includeMyTeams = isAllTeams || selectedTeams.includes('myteams');
+  const selectedOtherTeamIds = new Set(
+    selectedTeams.filter(teamId => teamId !== 'myteams')
+  );
+
+  const myTeams = includeMyTeams ? myTeamsInAll : [];
+  const otherTeams = isAllTeams
+    ? otherTeamsInAll
+    : [...otherTeamsInAll].filter(team => selectedOtherTeamIds.has(String(team.id)));
+
+  const visibleTeams = [...myTeams, ...otherTeams].filter(team => {
```

Comment:
this contained duplicated entries too

Reply:
ok this actually have to contain duplicates, otherwise the test "renders correct project with selected team" breaks.